### DISCUSSION TITLE
Trying to link to Globo free channels

### DISCUSSION THREAD

**Discussion by MateusAuri** [iptv-org/community#25] [Show and tell]
Globo, the brazilian TV network, broadcasts some of their channels for free on their Globoplay website, but you need to login to an account to use it.

If you inspect the page while watching a channel and search for m3u8 under the Network tab, you can find URLs to those streams that work on video players like MPC-BE or VLC, **however**:
• They have a 2 hour delay (unlike watching through the official globoplay website, which seems to be live)
• I'm not sure if the URLs are generated exclusively to the logged user
• Also not sure if the URLs are static or if they change and become unusable with time
• The URLs seem to be related to the closest local subsidiary, maybe they're automatically assigned based on your detected location

Here are the channels they will let you stream for free:
https://globoplay.globo.com/tv-globo/ao-vivo/6120663/
https://globoplay.globo.com/globonews/ao-vivo/7339101/
https://globoplay.globo.com/futura/ao-vivo/7420604/
https://globoplay.globo.com/ge-fast/ao-vivo/11134179/
https://globoplay.globo.com/receitas-fast/ao-vivo/11134191/
https://globoplay.globo.com/cbn-sp/ao-vivo/9182156/
https://globoplay.globo.com/cbn-rj/ao-vivo/9182126/

I'll refrain from sharing the direct m3u8 URLs for now, but you can easily log into an account and find them to test things out.

**Comment by pablo-fior** [comment]
> Globo, the brazilian TV network, broadcasts some of their channels for free on their Globoplay website, but you need to login to an account to use it.
> 
> If you inspect the page while watching a channel and search for m3u8 under the Network tab, you can find URLs to those streams that work on video players like MPC-BE or VLC, **however**: • They have a 2 hour delay (unlike watching through the official globoplay website, which seems to be live) • I'm not sure if the URLs are generated exclusively to the logged user • Also not sure if the URLs are static or if they change and become unusable with time • The URLs seem to be related to the closest local subsidiary, maybe they're automatically assigned based on your detected location
> 
> Here are the channels they will let you stream for free: https://globoplay.globo.com/tv-globo/ao-vivo/6120663/ https://globoplay.globo.com/globonews/ao-vivo/7339101/ https://globoplay.globo.com/futura/ao-vivo/7420604/ https://globoplay.globo.com/ge-fast/ao-vivo/11134179/ https://globoplay.globo.com/receitas-fast/ao-vivo/11134191/ https://globoplay.globo.com/cbn-sp/ao-vivo/9182156/ https://globoplay.globo.com/cbn-rj/ao-vivo/9182126/
> 
> I'll refrain from sharing the direct m3u8 URLs for now, but you can easily log into an account and find them to test things out.

Inspection the response from request pointed to de .m3u8 file I noticed that it returns a list of .ts files urls. I also noticed that the browser always loads the last .ts file from this list to show live stream. Can it be that when you read the .m3u8 stream from an iptv player it streams the first .ts file url causing the 2H delay? I didn't inspected the .js scripts but i believe that one of them is reading the lisst and loading the last .ts url, while iptv player read the first.
The .ts url streams an average of 2MB of data. if it realy is the channel stream, it is a very limited amount, it seems that this is true because the browser constantly request the list to the .m3u8 endpoint.
I realy don't know how iptv players work and I may be talking non sense, but it makes sense =)

Dando uma inspecionada na resposta retornada pela consulta ao arquivo .m3u8 reparei q ele retorna uma lista de url para arquivos .ts. Reparei tb que o browser sempre carrega o último item da lista para exibir a programação ao vivo. Será que ao importar o .m3u8 no player independente nao é carregado o primeiro arquivo .ts gerando o atrado de 2H? nao inspecionei os scripts .js, mas desconfio que através de um deles é feito o carregamendo do ultimo arquivo .ts da lista enquanto players independendes carreguem a partir do primeiro.
